docs/userguide/constraints.md

```diff
@@ -222,10 +233,123 @@ DiscreteLinkedParametersConstraint(
+second switch is on "right" respectively.
+```python
+from baybe.constraints import DiscreteDependenciesConstraint, SubSelectionCondition
+
+DiscreteDependenciesConstraint(
+    parameters=["Switch1", "Switch2"],  # the two parameters upon which others depend
+    conditions=[
+        SubSelectionCondition(selection=["on"]),     # values of Switch1 that activate the affected parameters
+        SubSelectionCondition(selection=["right"]),  # values of Switch2 that activate the affected parameters
+    ],
+    affected_parameters=[
+      ["Solvent", "Fraction"],  # parameters affected by "Switch1"
+      ["Frame1", "Frame2"]      # parameters affected by "Switch2"
+    ],
+)
+```
+
+An end to end example can be found [here](../../examples/Constraints_Discrete/dependency_constraints).
 
 ### ``DiscretePermutationInvarianceConstraint``
-Content coming soon...
+Permutation invariance is a property where combinations of values of multiple
+parameters do not depend on their order due to some symmetry in the experiment. 
+Suppose that we are creating a mixture which has up to three solvents, i.e. parameters 
+"Solvent 1", "Solvent 2", "Solvent 3".
+In this situation, all combinations from the following table would be equivalent, 
+hence the ``SearchSpace`` should effectively only contain one of them.
+
+|   | Solvent 1    | Solvent 2    | Solvent 3    |
+|---|--------------|--------------|--------------|
+| 1 | Substance 43 | Substance 3  | Substance 12 |
+| 2 | Substance 43 | Substance 12 | Substance 3  |
+| 3 | Substance 3  | Substance 12 | Substance 43 |
+| 4 | Substance 3  | Substance 43 | Substance 12 |
+| 5 | Substance 12 | Substance 43 | Substance 3  |
+| 6 | Substance 12 | Substance 3  | Substance 43 |
+
+```{note}
+Complex properties such as permutation invariance not only affect the search space, but 
+should ideally also constrain the surrogate model. For instance, the kernels in a 
+Gaussian process can be made permutation invariant to reflect this constraint, which 
+generally results in a better learning curve. Note, however, that not all surrogate 
+models provided by BayBE can easlily include such invariances. In these cases the 
+invariance is silently ignored, which will not be an issue, but just means these models 
+cannot benefit from a priori known constraints and invariances between parameters.
+```
+
+Let's add to the mixture example the fact that not only the choice of substance but also 
+their relative mixture factions are parameters "Fraction 1", "Fraction 2" and 
+"Fraction 3".
+This example also implies that the solvent parameters depend on their corresponding 
+fraction being ``> 0.0``, because in the case ``== 0.0`` the choice of solvent is 
+irrelevant. This models a campaign that allows "up to, but not necessarily, 
+three solvents".
+
+```{important}
+If some of the ``parameters`` of the ``DiscretePermutationInvarianceConstraint`` are 
```

Comment:
Why is parameters in backticks here? Feels to me that we should just use the normal word parameters here, as we do not specifically talk about the corresponding objects,

Reply:
hmm this is quite literally talking about the `parameters` object of `DiscretePermutationInvarianceConstraint`, so what should I do? Also in some of your other comments you seem to suggest using this also for other things like `conditions` which would be consistent with exactly this version here?

Reply:
Ahhh I simply misunderstood this here, so no need to change anything.